### DISCUSSION TITLE
Gallery extension doesn't work with (some) PNG or SVG files

### DISCUSSION THREAD

**Discussion by IngoBelka** [datenstrom/community#569] [Report a bug] [answered]
I have these lines on my site:

```
JPEG
[gallery *.*jpg photoswipe 150 150]

PNG
[gallery *.*png photoswipe 150 150]

Vektorgrafiken
[gallery *.*svg photoswipe 150 150]
```
With JPEG I get functional gallery while clicking an image.

With PNG or SVG files the gallery doen't work. The new layer with the functions for forward or backward and so on is not shown.

In the debug console (Firefox) I find this:
```
Uncaught TypeError: el.getAttribute(...) is null
    parseElements https://testserver/kristall/media/extensions/bundle-5d3ef7a108.min.js:28
    openPhotoSwipe https://testserver/kristall/media/extensions/bundle-5d3ef7a108.min.js:38
    onClickGallery https://testserver/kristall/media/extensions/bundle-5d3ef7a108.min.js:38
```
Why does it working with JPEG files and doesn't with the other ones? How can I solve this, if it is possible for me, to solve it?

**Comment by IngoBelka** [comment]
A deeper look into this problem led me to the `gallery.js` and there to this code block:
```
 size = el.getAttribute("data-size").split("x");
            item = {
                src: el.getAttribute("href"),
                w: parseInt(size[0], 10),
                h: parseInt(size[1], 10),
                caption: el.getAttribute("data-caption")
            };
```
May be it helps to find the problem.

**Comment by markseuffert** [comment]
Could be a problem with image detection. Can you provide some example images to reproduce it?

**Comment by IngoBelka** [comment]
Please take a look at this: https://www.grimmen.de/netzwerk/das-b%C3%BCndnis/gallerie

**Comment by markseuffert** [comment, marked as answer]
I updated the gallery extension, now it can handle undetected and broken images.

![Screenshot](https://user-images.githubusercontent.com/2429165/100849368-c296fd00-3482-11eb-8b70-a1e0664e8045.png)

For best results edit your SVG files. Use a numerical width and height, e.g. `<svg width="500" height="500" ...>`. Then there's no guess work for the software what the original image dimensions should be. Undetected SVG images might be shown with the wrong image dimensions. In any case the gallery should work, please try it out.

**Comment by IngoBelka** [comment]
Thank you very much Mark. It works perfectly now. And thank you for the hint with the size settings for SVG files.

**Comment by markseuffert** [comment]
Thank you for testing and giving feedback.

**Comment by IngoBelka** [comment]
Ok, all SVG files does have settings for width and height. But some have integer values and other floting point values. By which reason ever. Some are definend with px some without.
Firefox is showing for all SVG's the size values. But, as a small file explorer I'm using Tiny File Manager 2.4.3, and this tool doesn't recognize the dimensions, too. 
So, for me currently it is not clear, where the problem was. 
But nevertheless, it works now, that's the only importent point. :-)

**Comment by markseuffert** [comment]
Well, the SVG specification allows for a [wide range of formats and units](https://www.w3.org/TR/SVG11/types.html#DataTypeLength). Datenstrom Yellow supports only numerical width and height. No floating point values, no units, no bells and whistles. Undetected images will be shown with 500x500 pixels in the gallery, but this is just a fallback so that it works no matter what you throw at it.

**Comment by IngoBelka** [comment]
Oh man, this specification is horrorbly for each coder, I think. Not a program for SVG can use all  of thes rules - it's a question of a decision which and how many will realized. 

But we can close the issue now, can't we.
